projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -1156,29 +1159,34 @@ export class IgxTimePickerComponent implements
      */
     public openDialog(timePicker: IgxTimePickerComponent = this): void {
 
-        if (this.collapsed) {
-            this.collapsed = false;
-
+        if (this.toggleRef.collapsed) {
             let settings;
             if (this.mode === InteractionMode.Dialog) {
                 settings = this.overlaySettings || this._dialogOverlaySettings;
             }
 
             if (this.mode === InteractionMode.DropDown) {
                 settings = this.overlaySettings || this._dropDownOverlaySettings;
```

Comment:
you can remove _dialogOverlaySettings and assign it like this:
settings = this.overlaySettings;